src/lib/render-lit-html.ts

```diff
@@ -181,14 +180,19 @@ type AttributePartOp = {
 type CustomElementOpenOp = {
   type: 'custom-element-open';
   tagName: string;
-  ctor: any;
+  ctor: RenderableCustomElement;
+  staticAttributes: Map<string, string>;
 };
 
 /**
  * Operation to render a custom element, usually its shadow root.
  */
-type CustomElementRenderOp = {
-  type: 'custom-element-render';
+type CustomElementRenderChildrenOp = {
```

Comment:
Update doc and add one below. Why are they separate? Is it to allow the instance to reflect attributes before streaming them?